AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -3439,6 +3491,164 @@ function Debug-AzStorageAccountAuth {
             }
         }
 
+        if (!$filterIsPresent -or ($Filter -match "CheckUserRbacAssignment")) {
+            try {
+                $checksExecuted += 1
+                Write-Verbose "CheckUserRbacAssignment - START"
+
+                Request-ConnectAzureAD
+
+                $sidNames = @{}
+                $user = Get-NativeAdUser -Identity $UserName -Domain $Domain -ErrorAction Stop
+                $sidNames[$user.SID.Value] = $user.DistinguishedName
+
+                $groups = Get-NativeAdUserGroups -Identity $user.SID -Domain $Domain -ErrorAction Stop
+                $groups | ForEach-Object { $sidNames[$_.SID.Value] = $_.DistinguishedName }
+
+                # The user needs following role assignments to have the share-level access.
+                # Currently only three roles are defined, but new ones may be added in future,
+                # hence use a prefix to check.
+                # Storage File Data SMB Share Reader
+                # Storage File Data SMB Share Contributor
+                # Storage File Data SMB Share Elevated Contributor
+                $smbRoleNamePrefix = "Storage File Data SMB Share"
+                $smbRoleDefinitions = @{}
+                Get-AzRoleDefinition | Where-Object { $_.Name.StartsWith($smbRoleNamePrefix) } `
+                    | ForEach-Object { $smbRoleDefinitions[$_.Id] = $_ }
+                
+                $roleAssignments = Get-AzRoleAssignment -ResourceGroupName $ResourceGroupName `

```

Comment:
This switch might simplify the entire logic if it can be applied to our scenario. -ExpandPrincipalGroups.

https://docs.microsoft.com/en-us/powershell/module/az.resources/get-azroleassignment?view=azps-5.0.0

Reply:
Thanks for the tip! However I don't think much benefits of using -ExpandPrincipalGroups here for the reasons: 1. It is required to be used with -ObjectId, but we don't have it. 2. We are filtering to SMB only assignments anyway. 3. It doesn't help much on building the report below.

Reply:
Yes. I was thinking of getting OID using lookup but we still need to do role definition lookup and match that with what is assigned to the user. End result would be the same.